engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/bool/ToBoolNode.java

```diff
@@ -0,0 +1,37 @@
+package org.enso.interpreter.node.expression.builtin.bool;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import org.enso.interpreter.dsl.MonadicState;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.state.Stateful;
+
+@NodeInfo(description = "Helper node to handle Bool values that can also return DataflowErrors")
+public abstract class ToBoolNode extends Node {
```

Comment:
A convention we use is to call such throwing conversions `ExpectBoolNode`. `ToBoolNode` suggests you can convert anything.